Private/magda/navbar

madziarq opened the pull request:
Do dopracowania jeszcze skalowanie po zmniejszeniu okna i pod mobilkę (z jakiegoś powodu mi ikonki z FontAwesome nie działają) oraz wyglądu buttona w zależności od 'strony' na której się jest (ale to się dorobi później jak już będą stronki). Dodałam też fonty i trochę dopracowałam stylowanie buttona (również na widoku strony logowania itp.).

madziarq committed:
Added fonts and slightly changed styling of components

madziarq committed:
Added fonts and slightly changed styling of components

madziarq committed:
Corrected font size and button's size

madziarq committed:
Navbar's first version

madziarq committed:
Created Navbar's view for testing

Abbion commented on app/src/Css/Navbar.css:
```diff
@@ -0,0 +1,36 @@
+    align-items: center;
+
+    position: absolute;
+    width: 1920px;
```
To lepiej w procętach. Niech zależy od szerokości okna.

Abbion replied:
To jest stare.

Abbion commented:
Mergujemy gdy dany komponent jest w 100% sprawny. Tu brakuje tych ikonek i przy widoku mobilnym wszystko się rozjeżdża.

Abbion commented:
Osoby, które będą robiły strony po zalogowaniu tymczasowo dadzą sobie pustego diva tak tylko aby za symbolizować sobie navbar.

Abbion commented:
Dodatkowo jakbyś mogła dodać prosty output po kliknięciu w każdy z przycisków.

madziarq committed:
Navbar attempt #2

Adjusted scalling, added icons (still don't work idk why), tried to add drop-down menu but not successfully.

madziarq committed:
Navbar attempt #2

Adjusted scalling, added icons (still don't work idk why), tried to add drop-down menu but not successfully.

madziarq committed:
Navbar attempt #3

Almost final version - drop-down menu works (needs better button placement), added working icons.

madziarq committed:
Navbar attempt #3

Almost final version - drop-down menu works (needs better button placement), added working icons.

madziarq committed:
Navbar - final version

madziarq committed:
Navbar - final version

madziarq commented:
Navbar sam w sobie gotowy - skalowanie i rozwijane menu działa, ikonki też wreszcie udało mi się dodać. Jedyne czego nie zdążyłam jeszcze dopracować to outputów bo nie wiem jeszcze jak najlepiej ogarnąć to żeby zmieniało content w kontenerze a nie całą stronę. Jeżeli u Was też zadziała wszystko i nic nie trzeba poprawiać/dodawać na ten moment to można mergować.

madziarq commented:
Jeśli chodzi o konflikty, w App.js są tylko zaimplementowane widoki więc to nie ma większego znaczenia; w index.css jedyne co zrobiłam to już dodałam fonty Adam zamiast Roboto, html nie tykałam; w BasicComponents.css zmieniłam jedynie rozmiary przycisku żeby lepiej wyglądały na navbarze i widoku rejestracji (nie wiem jak wygląda na innych widokach, i tak mogę dostosować stylowanie buttona do samego navbara).

Abbion committed:
rebase

Abbion commented on app/src/Components/MenuItems.js:
```diff
@@ -0,0 +1,23 @@
+import '../Css/BasicComponents.css'
```
Lepiej jak przeniesiesz to wszystko do Navbar.js

Abbion commented on app/src/Components/Navbar.js:
```diff
@@ -0,0 +1,29 @@
+        return(
+            <nav className="NavbarItems">
+                <ul className={this.state.clicked ? "nav-menu-active" : "nav-menu"}>
+                    {MenuItems.map((button, index) => React.cloneElement(button, {key: index}))}
```
Mamy tylko 4 zakładki lepiej wpisać je tu ręcznie.

Abbion commented on app/src/Css/Navbar.css:
```diff
@@ -0,0 +1,100 @@
+
+.Button {
+    width: calc(45px + 12vw);
+    font-size: calc(10px + 0.88vw);
```
Daj jakąś stałą wartość

Abbion commented on app/src/Css/Navbar.css:
```diff
@@ -0,0 +1,100 @@
+.nav-menu {
+    display: grid;
+    grid-template-columns: repeat(4, auto);
+    grid-gap: calc(16px + 2vw);
```
Nie używaj dynamicznego obliczania dla takich rzeczy, lepiej używać procentów i stałych np. pixeli.

Abbion commented on app/src/Components/Navbar.js:
```diff
@@ -0,0 +1,29 @@
+import MenuBars from "../Graphics/Icons/MenuBars.png"
+import MenuXmark from "../Graphics/Icons/MenuXmark.png"
+
+class Navbar extends Component {
```
Nie używamy klas, to stara technika i nie powinno się już tak pisać. Lepiej użyć funkcji tak jak w innych komponentach.

Abbion commented:
Dodatkowo jest jakiś problem z grubością czcionki. Na release jest cieńsza niż u Ciebie. Czcionkę zmieniliśmy z Adam na Roboto, bo jest bardziej czytelna.

Abbion commented:
Jeszcze jedna rzecz. Jak widok mobilny nie powinien mieć paska na całej długości a bardziej kwadratową ikonkę w prawym górnym rogu.

madziarq committed:
Navbar - (hopefully) final version

- Changed form of Navbar component from class to functional,
- Packed all buttons in Navbar.js without using useless MenuItems.js file,
- Corrected styling, repaired (somehow) font weight, changed scaling to more static, changed mobile version of menu to match with prototype on Figma,
- Added LogoutButton (forgot about it before),
- Added styling for buttons, that are already clicked

madziarq commented:
Udało mi się przepisać Navbar z class na functional component. Wywaliłam plik MenuItems.js i wsadziłam wszystko w jednym miejscu, przy okazji dodałam Logout button bo mi to umknęło wcześniej. Naprawiłam (jakoś) problem z czcionką a przynajmniej na potrzeby Navbara i widoku, z którego korzystałam. Dodałam też stylowanie dla klikniętego przycisku (tak jak na prototypie, jak jest się na danym widoku to przycisk jest ciemny). Skalowanie bardziej statyczne tak jak @Abbion powiedziałeś, no i też menu na mobilkę jest już bardziej jak w prototypie (niestety z jakiegoś powodu nie jestem w stanie ułożyć ich dokładnie jak na prototypie, to najlepsze co udało mi się osiągnąć).

madziarq commented:
Zapraszam do sprawdzenia, mam nadzieję że już nie będzie wiele do poprawy :')

Abbion committed:
Small fixes

Abbion committed:
rebase

Abbion commented:
Trochę rzeczy poprawiłem, dla lepszego 'user experience'. No jest trochę powtarzającego się kodu, dało by się skrócić css o tak 50 linijek. Poza tym wszystko ok i bardzo ładna animacja otwierającego się menu. Był też duży problem z mergem, ale chyba wszystko działa dobrze.
